tests/Integration/Database/EloquentModelUuidCastingTest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Illuminate\Tests\Integration\Database\EloquentModelUuidCastingTest;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Tests\Integration\Database\DatabaseTestCase;
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidInterface;
+
+class EloquentModelUuidCastingTest extends DatabaseTestCase
+{
+    protected function defineDatabaseMigrationsAfterDatabaseRefreshed()
+    {
+        Schema::create('test_model1', function (Blueprint $table) {
+            $table->increments('id');
+            $table->uuid();
+        });
+    }
+
+    public function testUuidsAreCustomCastable()
+    {
+        $user = TestModel1::create([
+            'uuid' => '41828d7a-997c-4190-976a-ff2cf61f0b52',
```

Comment:
I'm confused. I thought the point of casts were you provide an object and it normalizes it down to a string for storage. But here you seem to pass a raw string to the `create` method. Contrast this to other casts like dates where you would pass a DateTime instance here - not a string.

Reply:
> I'm confused. I thought the point of casts were you provide an object and it normalizes it down to a string for storage. But here you seem to pass a raw string to the `create` method. Contrast this to other casts like dates where you would pass a DateTime instance here - not a string.

This is actually a mirror of the date and datetime functionality, where it works both ways. This way you can seed from raw data without having to create any objects.